src/Children.js

```diff
@@ -0,0 +1,101 @@
+import * as Utility from './Utility.js'
+import * as Element from './Element.js'
+
+export default {only: only, count: count, filter: filter, find: find, map: map, forEach: each, toArray: array}
+
+/**
+ * @throws {Error} if not a single valid element
+ * @param {*} value
+ * @return {object?}
+ */
+export function only (value) {
+	return Element.valid(value) ? value : Utility.invariant('Expected to receive a single element!')
+}
+
+/**
+ * @memberof Children
+ * @param {*} value
+ * @return {number}
+ */
+export function count (value) {
+	var children = {length: 0}
+
+	Utility.each(function (value, index, children) {
+		children.length++
+	}, value, 0, children)
+
+	return children.length
+}
+
+/**
+ * @param {*} value
+ * @param {function} callback
+ * @return {Array}
+ */
+export function filter (value, callback) {
+	var children = []
+
+	Utility.each(function (value, index, children) {
+		if (callback(value, index, children)) {
+			children.push(value)
+		}
+	}, value, 0, children)
+
+	return children
+}
+
+/**
+ * @param {*} value
+ * @param {function} callback
+ * @return {*?}
+ */
+export function find (value, callback) {
+	var children = {value: null}
+
+	Utility.each(function (value, index) {
+		if (callback(value, index, children)) {
+			return children.value = value
+		}
+	}, value, 0)
+
+	return children.value
+}
+
+/**
+ * @param {*} value
+ * @param {function} callback
+ * @return {Array}
+ */
+export function map (value, callback) {
+	var children = []
+
+	Utility.each(function (value, index, children) {
+		children.push(callback(value, index, children))
```

Comment:
passing `children` to the callback seems inorrect. I'd think (if needs to exist at all) the third argument would be the original `value` passed into the function (a la lodash)

Reply:
The reason children is being passed to `each` is to avoid creating a closure, but the primary reason was due to possibility of `Element.from` being a callback, since we use native `Array.map` with `Element.from`  it makes sense to respect the arity of `Array.map` given `Element.from` can be a hot function.

Reply:
Arity-wise passing the original `value` would achieve the same but also match `Array.map()` usage patterns. Though if `value` is not an array maybe would cause the optimizer to do extra work... I guess it's fine then, though this level of micro-optimization slightly affects readability